### PR TITLE
Feat: 1462 attachment form

### PR DESCRIPTION
**Pull request by BCerki**
card: https://app.zenhub.com/workspaces/climate-action-secretariat-60ca4121764d710011481ca2/issues/gh/bcgov/cas-cif/1462

Note: attachment table filtering and sorting previously wasn't working: https://app.zenhub.com/workspaces/climate-action-secretariat-60ca4121764d710011481ca2/issues/gh/bcgov/cas-cif/809

### CODE REVIEW

**Review by Sepehr-Sobhani — CHANGES_REQUESTED** [review]
Looks great 🤩 some minor changes are required.

File: app/components/Form/ProjectAttachmentsForm.tsx [hunk 1]
```diff
@@ -124,19 +140,25 @@ export function ProjectAttachments({
       )}
       <Table
         paginated
-        totalRowCount={projectRevision.projectAttachmentFormChanges.totalCount}
+        totalRowCount={projectAttachmentFormChanges.totalCount}
         filters={tableFilters}
-        pageQuery={AttachmentsQuery}
```
Comment (BCerki) [inline app/components/Form/ProjectAttachmentsForm.tsx:129]: I don't have a page query anymore now that attachments are a component, so I can't use this prop. Not sure if I need to do something to replace it? This is what it was doing:

```/**
   * The top-level query used by the page rendenring the table.
   * If provided, when the table filters, ordering or pagination are updated,
   * the table will reload the query before updating the router,
   * preventing a render with the Suspense fallback while the new data is being fetched.
   */
  pageQuery?: GraphQLTaggedNode;

File: app/lib/relay/client.ts
[the remark this answers is not in the document]
Comment (BCerki) [reply app/lib/relay/client.ts:1]: There's a batch error if the middleware is in the old order. We hit it on CIIP too

File: schema/test/unit/mutations/discard_project_attachment_form_change_test.sql [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+  $$
+    with record as (
+    select row(form_change.*)::cif.form_change
+    from cif.form_change where id=2
+    )
+    select
+      id,
+      operation,
+      project_revision_id
+    from cif.discard_project_attachment_form_change(2)
```
[the remark this answers is not in the document]
Comment (Sepehr-Sobhani) [reply schema/test/unit/mutations/discard_project_attachment_form_change_test.sql:66]: I tried removing this part and test passed, Am I doing something wrong?
```
    with record as (
    select row(form_change.*)::cif.form_change
    from cif.form_change where id=2
    )
```

[the remark this answers is not in the document]
Comment (BCerki) [reply schema/test/unit/mutations/discard_project_attachment_form_change_test.sql:66]: Probably I had something wrong originally. Thanks, I'll simplify it!

**Comment by Sepehr-Sobhani** [comment]
@BCerki Just another question, Do we also need to have `latestCommittedFormChangesFor` in the project attachment form summary too?

**Comment by BCerki** [comment]
> @BCerki Just another question, Do we also need to have `latestCommittedFormChangesFor` in the project attachment form summary too?

That's part of https://github.com/bcgov/cas-cif/issues/1680

**Pull request merged**